src/Payment/OrderLines.php

```diff
@@ -64,12 +64,48 @@ public function order_lines($order, $voucherDefaultCategory)
         $this->process_shipping();
         $this->process_fees();
         $this->process_gift_cards();
+        $this->process_missmatch();
 
         return  [
             'lines' => $this->get_order_lines(),
         ];
     }
 
+    private function process_missmatch()
+    {
+        $orderTotal = $this->order->get_total();
+        $orderTotalRounded = round($orderTotal, 2);
+        $linesTotal = array_sum(array_map(function ($line) {
+            return $line['totalAmount']['value'];
+        }, $this->order_lines));
+        $orderTotalDiff = $orderTotalRounded - $linesTotal;
+        if (abs($orderTotalDiff) > 0) {
```

Comment:
@mmaymo it looks good to me, but I have one question. There is one scenario that could cause problems. Maybe you know what happens. For example:
`$orderTotalRounded - $linesTotal = 0.001`
`$this->order_lines[] = $missmatch;`
Mollie may round this to 0.00 and we have a surcharge with 0.00. Can that happen?

Reply:
Good point, I'll check it

Reply:
Yeah, seems if the rounding makes the diff line send a 0 value there is no problem. The transaction went through without error.

Reply:
Great, thank you :)